bundle-workflow/src/s3_utility.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
I had made a ton of comments on this file in a different PR, I think in #367 but I am no longer sure. Those needs to be addressed first.

Reply:
Yes that is the PR that introduces the S3 utility. I merged it in my branch to call the download_folder method from #367, so it is showing up in my PR.

Reply:
Ok, that code is not ready to be merged anywhere.